<a href="https://colab.research.google.com/github/Meowmixforme/OH-Deer/blob/main/Notebooks/01_Red_Deer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

First let's import the database and take a look at it's structure

In [1]:
import pandas as pd

df = pd.read_csv("records-2025-12-28.csv")

df.head()
df.info()
df.describe(include='all')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28964 entries, 0 to 28963
Data columns (total 54 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   NBN Atlas record ID                 28964 non-null  object 
 1   Occurrence ID                       28964 non-null  object 
 2   Licence                             28964 non-null  object 
 3   rightsHolder                        28465 non-null  object 
 4   Scientific name                     28964 non-null  object 
 5   Taxon author                        28964 non-null  object 
 6   Name qualifier                      0 non-null      float64
 7   Common name                         28964 non-null  object 
 8   Species ID (TVK)                    28964 non-null  object 
 9   Taxon Rank                          28964 non-null  object 
 10  Occurrence status                   28964 non-null  object 
 11  Event Date                          23444

,NBN Atlas record ID,Occurrence ID,Licence,rightsHolder,Scientific name,Taxon author,Name qualifier,Common name,Species ID (TVK),Taxon Rank,...,Family,Genus,OSGR 100km,OSGR 10km,OSGR 2km,OSGR 1km,Country,State/Province,Vitality,Public resolution (m)
count,28964,28964,28964,28465,28964,28964,0.0,28964,28964,28964,...,28964,28964,28601,28576,22188,21463,28957,28866,3470,7.0
unique,28964,28924,4,120,1,1,NaN,1,1,1,...,1,1,48,1560,4768,6971,2,4,5,NaN
top,ffff73d1-3181-4983-b2c6-153772b6640f,24995634,CC-BY-NC,North-East Scotland Biological Records Centre,Cervus elaphus,"Linnaeus, 1758",NaN,Red Deer,NBNSYS0000005143,species,...,Cervidae,Cervus,NN,NJ01,SK53J,SK5339,United Kingdom of Great Britain and Northern I...,Scotland,alive,NaN
freq,1,2,17102,8298,28964,28964,NaN,28964,28964,28964,...,28964,28964,4161,1815,484,398,28954,19432,2651,NaN
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2000.0
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2000.0
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2000.0
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2000.0
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2000.0


There are 28,964 rows and from a visual inspection we can see that:

- around 5,500 rows have no Event Date
- around 250 rows have no coordinates
- some records span date ranges
- some are duplicates from multiple datasets

First let's filter the record to only include valid spatial and temporal records.

In [2]:
df = pd.read_csv("records-2025-12-28.csv")

# Parse dates
df['Event Date'] = pd.to_datetime(df['Event Date'], errors='coerce')

# Keep only rows with usable coordinates AND a valid date
df_clean = df.dropna(subset=['Latitude (WGS84)', 'Longitude (WGS84)', 'Event Date'])

len(df_clean)

23422

Now we have 23422 records we should standardise column names

In [3]:
df_clean = df_clean.rename(columns={
    'Latitude (WGS84)': 'lat',
    'Longitude (WGS84)': 'lon',
    'Coordinate Uncertainty (m)': 'uncertainty',
    'Event Date': 'date',
    'Scientific name': 'scientific_name',
    'Common name': 'common_name'
})

Checking the count by year

In [6]:
pd.set_option('display.max_rows', None)
df_clean['year'].value_counts().sort_index()

,count
year,
1705,1
1736,135
1829,1
1830,1
1860,7
1880,1
1888,1
1900,2
1905,24


We can see Sparse, scattered early records from 1705 to around 1950.

A small rise in the late 1950s–1960s.

A huge jump in 1970 and then stable, dense data from 1980 onward.

There is a massive recorder boom in the 2000s–2020s, likely from citizen science contributions.




